### PR TITLE
Remove JavaScript layer and rework API in SPI module

### PR DESCRIPTION
**Pull request by LaszloLango**
Related issue: #1367

IoT.js-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: src/modules/nuttx/iotjs_module_spi-nuttx.c [hunk 1]
```diff
@@ -22,10 +24,42 @@
...
+  struct spi_dev_s* spi_dev;
+};
+
+void iotjs_spi_create_platform_data(iotjs_spi_t* spi) {
+  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_spi_t, spi);
+
+  _this->platform_data = IOTJS_ALLOC(iotjs_spi_platform_data_t);
+  _this->platform_data->bus = -1;
+  _this->platform_data->cs_chip = 0;
+  _this->platform_data->cs_chip = 0;
```
Comment (bzsolt) [inline src/modules/nuttx/iotjs_module_spi-nuttx.c]: Duplicated line.

File: test/run_pass/test_spi_buffer_sync.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/* Copyright 2016-present Samsung Electronics Co., Ltd. and other contributors
```
Comment (bzsolt) [inline test/run_pass/test_spi_buffer_sync.js]: 2018

**Comment by LaszloLango** [comment]
@bzsolt @glistening, thanks for the review. I've updated the PR.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**